### DISCUSSION TITLE
convert doesn't work after update from ImageMagick-6.8.6-8

### DISCUSSION THREAD

**Discussion by heinzelmann2** [ImageMagick/ImageMagick#3971] [Help]
Hello,
I had ImageMagick-6.8.6-8 running on a computer for many years. Since the computer is now broken I have set up a new Win10 with the latest ImageMagick version.
Now I noticed that the Convert command as I used it back then does not seem to work anymore.
convert.exe 122850.TIF -resize 3307x3307 +antialias -font c:\windows\fonts\arial.ttf -pointsize 90 -gravity west -type Grayscale -draw "translate 0,0 rotate -90 text 80,100 ""Job: 5025761 / AC7X""" -rotate "-90>" -density 300x300 +compress 122850_2.TIF

A TIF file (approx. 200KB) is read and modified. A text is drawn on the image.
With the old version of Convert the command worked great and the TIF could be opened afterwards.
With the new version there is no error message but the image has suddenly almost 8MB and cannot be opened anymore.

What has changed or what should my command be?

Thanks

**Comment by fmw42** [comment]
For ImageMagick 7, use magick not convert.  To use convert, you would have needed to install the legacy components with the install. But using convert would give IM 6 syntax. You want to use the newer IM 7, with mostly the same syntax, but using magick command rather than convert.

**Comment by snibgo** [comment]
> ... with the latest ImageMagick version.

The "latest version" means different things to different people. Please paste here the text output from "convert -version".

Your command works fine for me, with IM 6.9.9-50 on Windows 8.1. With v7.1.0-4, using "magick", the font filename needs to have either forward slashes, or doubled backslashes.

> ... but the image has suddenly almost 8MB ...

If your input was compressed then the output, which isn't compressed, may be larger. And if your input has less than 3307x3307 pixels, the output will have more pixels.

> ... and cannot be opened anymore.

By what programs? What happens instead? Any error messages?

**Comment by heinzelmann2** [comment]
Thank you for the help.
But now I have the next problem!
First here are the informations about my installation.
My old version of convert.exe is ImageMagick 6.8.6-8 2013-08-04 Q16
My new version of convert.exe is ImageMagick 7.1.0-2 Q16 x64 2021-06-25
I have installed the portable version of ImageMagick-7.1.0-2-portable-Q16-x64.zip

I want to upload the converted TIF into SAP-System. The problem is that SAP only accept a special TIF format.
I get the following error in SAP:
- TIFF format error: No baseline TIFF 6.0 file
- Error: BitsPerSample tag has length <> 1 or 3

With the old version of the convert.exe it works, but with the new version or with magick.exe it doesn't work. 
I have no idea what I have to change, so I get a SAP conform TIF file.

In the attached ZIP, you will find my files.
[122850.zip](https://github.com/ImageMagick/ImageMagick/files/6862636/122850.zip)

12850.tif is the original file with no special text on it
12850_1.tif is the file converted with the old version of convert.exe and can be uploaded to SAP
12850_3.tif is the file converted with the new magick.exe and can **not** be uploaded to SAP
12850_4.tip is the same as 12850_3. But I have opened it with IRFANVIEW and saved it again. This can be uploaded to SAP

Any idea?
Or if you have an proposal how I can better structure my command for converting and stamping the TIF please let me know.

Please excuse my bad english! 

Thanks

**Comment by snibgo** [comment]
The error is:
```
BitsPerSample tag has length <> 1 or 3
```
If we use "exiftool 122850-1.tif" etc on the files, we find:

122850-1.tif Bits Per Sample : 1
122850-3.tif Bits Per Sample : 1 1 1 1
122850-4.tif Bits Per Sample : 4

So "-1" has one channel of one bit, "-3" has four channels of one bit, and "-4" has one channel of 4 bits.

Your input file 122850.TIF has one channel of one bit, but with compression "T4/Group 3 Fax", and WhiteIsZero. If we do a simple:
```
magick 122850.TIF x.tif
```
... with IM v7.1.0-4 then exiftool says x.tif is one channel of one bit, with no compression, and BlackIsZero. This seems to be what you want, but is different to what you get with IM v7.1.0-2. So try "-colorspace Gray" to reduce the channels to one:
```
magick 122850.TIF -colorspace Gray x2.tif
```
